### ISSUE TITLE
How to get the actual value of multiple_checkbox field type

### ISSUE THREAD

**Issue by charles890123** [elegantthemes/create-divi-extension#40]
Hello.
Just a simple question. Can be a possibility to use a multiple_checkbox field type and get the actual values of the checked checkbox instead of a string like on||on|on|on|. 
I am getting the values from a query and assign to the multiple_checkbox field on the options parameter. Is there any other field type that I can use for this purpose?
Thanks,

**Comment by etstaging** [comment]
### QA Checklist
<strong>Reporter</strong>
- [ ] Helpful title used &nbsp;<img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Helpful title format:*** `Scope(s) :: One Sentence To Describe The Issue`
  * ***Example:*** `VB :: Floating Toolbar :: Jumps To Top Of Page`
- [ ] Detailed description <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Description should usually be 150+ characters.***
- [ ] Steps to reproduce included <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
- [ ] Support Ticket URL included
- [ ] Image and/or gif

<strong>Developer</strong>
- [ ] Steps to reproduce proven solved
- [ ] Test included and coverage acceptable and all Tests passing?
- [ ] PR description has layman's description of problem and solution and "Areas Affected" listed/explained?
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Estimated points on issue
- [ ] Proper code documentation and conventions?
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Peer Review</strong>
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Steps to reproduce proven solved
- [ ] Affected areas are tested and ok
- [ ] Proper code documentation and conventions
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Quality Assurance</strong>
- [ ] All tests passed on CircleCI.
- [ ] Steps to reproduce passed, the fix works.
- [ ] No errors in console or output 
- [ ] Affected areas are tested and ok
- [ ] The fix does not change a customer's website in any surprising or unintended way
  - [ ] If Divi's design <em>is</em> adjusted, settings are migrated gracefully and invisibly

<strong>Design Review</strong>
- [ ] UI is to spec
- [ ] UX is to spec
- [ ] Design is to spec
- [ ] No surprising visual changes to a customer's website will be experienced after updating
  - [ ] Any/all visual changes are migrated gracefully

<strong>Dev Review</strong>
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated? 
- [ ] i18n used if applicable
- [ ] Implementation acceptable, DRY, Scaleable, future safe
- [ ] Best practices used, cleverness and grokage, ES6 use, etc.

**Comment by lots0logs** [comment]
Hi, you can determine what each `on`|`off` corresponds to by using their index positions (they will always be in the same order as defined in the settings definition). Let me know if that helps :smiley:

**Comment by xxtesaxx** [comment]
Having the possibility to set a flag and get an array of the values which are checked rather than the on||on| list would be superb! I ran into this issue in 3.0.x and had to give up on an module idea. The only way to solve it would have been to write my own controls.

**Comment by lots0logs** [comment]
So, we're currently using the field type `categories` for this. However, I don't want to document it because its current API doesn't make much sense. Once we fix it, we'll document it. That being said, here is an example of how you can use it today:

```php
'include_categories'   => array(
	'label'            => esc_html__( 'Include Categories', 'et_builder' ),
	'type'             => 'categories',
	'renderer_options' => array(
		'use_terms'    => true,
		'term_name'    => 'product_cat',  // This is actually the taxonomy slug
	),
	'description'      => esc_html__( 'Choose which categories you would like to include.', 'et_builder' ),
	'toggle_slug'      => 'main_content',
),
```

**Comment by xxtesaxx** [comment]
I'm sure charles890123 already knows that and how to use it but that only works for terms of taxonomies :) 

Imagine for example you want to let the user select post types or maybe some other value which the user can define somewhere in the backend. Once a new value is added, it might mess with the selection. For example you have the post types "Artworks" and "Paintings" and now you add "Books". "on|on" becomes "on|on|" but instead of Artworks and Paintings, you now have Artworks and Books selected.

That is why we would need a way to let the multiple checkboxes field (or another new control) return an array or comma separated list of slugs. Somthing like this:

    'multi_select_test'   => array(
        'label'            => esc_html__( 'Select Multiple Boxes', 'et_builder' ),
        'toggle_slug'      => 'main_content',
        'type' => 'multiple_checkboxes',
        'options' => array(
            'hello' => esc_html__('Hello', 'et_builder'),
            'world' => esc_html__('World', 'et_builder'),
        ),
        'return_slugs' => true,
    ),

And the return value would be one of those:

    "hello,world"   
    "hello|world"
    array(hello, world)

**Comment by lots0logs** [comment]
Yeah, I agree that would make much more sense than how it works today. We're going to replace it with a more robust solution soon. Most likely at the same time we launch support for custom fields in the VB.

**Comment by charles890123** [comment]
Good Day @lots0logs Any updates with this.

**Comment by odoremieux** [comment]
Did you gave up on us?

**Comment by lots0logs** [comment]
No news on this, sorry! It has gotten pushed back due to other priorities. It's still planned though.

**Comment by odoremieux** [comment]
Any time frame?

**Comment by lots0logs** [comment]
No, I'm afraid not at this time. Sorry!

**Comment by mrkenng** [comment]
Any good news on this? :D

**Comment by easterncoder** [comment]
Hate to bump an old thread but any updates on this?

**Comment by xxtesaxx** [comment]
Just today I saw that apparently we now can create our own custom fields by providing a react component. You can now easily implement your own multi-select. You could for example use select2 or simply use buttons and also style them however you desire.

**Comment by easterncoder** [comment]
@xxtesaxx yeah, i was advised to go that route as well.

**Comment by gonzalesc** [comment]
> Just today I saw that apparently we now can create our own custom fields by providing a react component. You can now easily implement your own multi-select. You could for example use select2 or simply use buttons and also style them however you desire.

Hi. Do you know any link about this, please?

**Comment by xxtesaxx** [comment]
Simply create a new divi extension with npx create-divi-extension and check the sample in includes/fields. It's just another react component but I believe the value is stored as plain text so what I usually do is store more complex values as stringified json

**Comment by xxtesaxx** [comment]
I also recommend to console.log or debug the field in render() to see all the props you have available. you can pass a lot of stuff via the field definition from php to jsx

**Comment by aaronmeder** [comment]
I could also use a better multi checkbox field 🙋🏻‍♂️

**Comment by divibooster** [comment]
I know this thread is old, but there still doesn't seem to be a native solution for this and I couldn't find an example jsx component, so I wrote my own and posted it here:

[Divi Custom Field Type: Multi-checkboxes with IDs](https://divibooster.com/divi-custom-field-type-multi-checkboxes-with-ids/)

Hope it's of use to someone :)

**Comment by seebeen** [comment]
It is. I'll backport it to my plugin.

Thanks a lot, bratan
